myfitnesspal/client.py

```diff
@@ -475,31 +493,45 @@ def get_date(self, *args, **kwargs) -> Day:
             )
         document = self._get_document_for_url(
             self._get_url_for_date(
-                date, kwargs.get("username", self.effective_username)
+                date, kwargs.get("username", self.effective_username), kwargs.get("friend_username")
             )
         )
+        if "diary is locked with a key" in document.text_content():
+            raise Exception("Error: diary is locked with a key") 
```

Comment:
Under what circumstances do we get this message?

We might also add:
```suggestion
        if "diary is locked with a key" in document.text_content():
            raise Exception("Error: diary is locked with a key")
        if kwargs.get("friend_username") is not None and "user maintains a private diary" in document.text_content():
            raise Exception(f"Error: Friend {kwargs.get('friend_username')}'s diary is private.")
```